### PR TITLE
Fix failing CI for sqlite

### PR DESCRIPTION
**Pull request by qnkhuat**
Fix the failing tests on sqlite introduced by #26474

### CODE REVIEW

File: test/metabase/query_processor_test/date_bucketing_test.clj [hunk 1]
```diff
@@ -753,9 +753,9 @@
                           (map second)
                           (take 3)))))
             (testing "make sure all drivers returns the same week column"
-              (is (= ["2019-01-01T00:00:00Z"
-                      "2019-01-08T00:00:00Z"
-                      "2019-01-15T00:00:00Z"]
+              (is (= (case driver/*driver*
+                       :sqlite ["2019-01-01 00:00:00" "2019-01-08 00:00:00" "2019-01-15 00:00:00"]
+                       ["2019-01-01T00:00:00Z" "2019-01-08T00:00:00Z" "2019-01-15T00:00:00Z"])
```
Comment (calherries) [inline test/metabase/query_processor_test/date_bucketing_test.clj:758]:
```suggestion
              (is (= (cond-> ["2019-01-01T00:00:00Z" "2019-01-08T00:00:00Z" "2019-01-15T00:00:00Z"]
                       (= driver/*driver* :sqlite) (comp u.date/format-sql t/local-date-time t/zoned-date-time))
```

**Review by calherries — APPROVED** [review]
Approved for the sake of CI, but ideally we would have a formatting function we could reuse for sqlite everywhere

**Comment by flamber** [comment]
@qnkhuat Remember to backport this and #26474, but perhaps it's easier to just create a single PR manually?

**Comment by qnkhuat** [comment]
yes, I'll create a manual back port that has both changes tmr

**Pull request merged**